examples/geometry_optimization.jl

```diff
@@ -52,33 +57,32 @@ function fg!(F, G, x)
         G .= -[grad[1][1]; grad[1][2]]
     end
     scfres.energies.total
-end
+end;
 
 # Now, we can optimize on the 6 parameters `x = [x1, y1, z1, x2, y2, z2]` in
-# reduced coordinates, using `LBFGS()` and starting from `x0`. By
+# reduced coordinates, using `LBFGS()`, the default minimization algorithm
+# in Optim. We start from `x0`, which is a first guess for the coordinates. By
 # default, `optimize` traces the output of the optimization algorithm during the
 # iterations. Once we have the minimizer `xmin`, we compute the bond length in
 # cartesian coordinates.
 
-x0 = vcat(invlattice*[0., 0., 0.], invlattice*[1.4, 0., 0.])
+x0 = vcat(lattice \ [0., 0., 0.], lattice \ [1.4, 0., 0.])
 xres = optimize(Optim.only_fg!(fg!), x0, LBFGS(),
                 Optim.Options(show_trace=true, f_tol=tol))
 xmin = Optim.minimizer(xres)
 dmin = norm(lattice*xmin[1:3] - lattice*xmin[4:6])
-sdmin = @sprintf "%.3f" dmin
-println("\nBond length minimizing the energy for Ecut=$(Ecut): ", sdmin, " Bohr");
+@printf "\nOptimal bond length for Ecut=%.2f: %.3f Bohr\n" Ecut dmin
 
-# We used here very rough parameters to generate the example,
-# but setting `Ecut` to 10 Ha yields a bond length of 1.523 Bohr.
-# As a comparison, ABINIT with the exact same lattice constant, Ecut = 10,
+# We used here very rough parameters to generate the example and
+# setting `Ecut` to 10 Ha yields a bond length of 1.523 Bohr.
+# As a comparison, ABINIT with the exact same lattice constant, `Ecut = 10`,
 # LDA with Teter parametrization and a pseudopotential from the
 # Goedecker-Hutter-Teter table gives a bond length of 1.522 Bohr
 # (cf. the [ABINIT tutorial](https://docs.abinit.org/tutorial/base1/)).
 #
 # !!! note "Degrees of freedom"
 #     We used here a very general setting where we optimized on the 6 variables
-#     representing the position of the 2 atoms and it can be then easily extended
+#     representing the position of the 2 atoms and it can be easily extended
 #     to molecules with more atoms (such as ``H_2O``). In the particular case
-#     of ``H_2``, there is in fact only one unknown which is the bond length ``d``,
-#     and we could have used 1D optimization algorithms to minimize an energy
-#     functional built just as a function of the variable ``d``.
+#     of ``H_2``, we could use only one degree of freedom which, in this case,
```

Comment:
I would really mention that it is just the *internal* degree of freedom that you need to consider.